tools/dpip/addr.c

```diff
@@ -221,29 +226,34 @@ static int addr_do_cmd(struct dpip_obj *obj, dpip_cmd_t cmd,
 
     switch (conf->cmd) {
     case DPIP_CMD_ADD:
+        param.ifa_ops = INET_ADDR_ADD;
         return dpvs_setsockopt(SOCKOPT_SET_IFADDR_ADD, &param, sizeof(param));
     case DPIP_CMD_DEL:
+        param.ifa_ops = INET_ADDR_DEL;
         return dpvs_setsockopt(SOCKOPT_SET_IFADDR_DEL, &param, sizeof(param));
     case DPIP_CMD_SET:
+        param.ifa_ops = INET_ADDR_MOD;
         return dpvs_setsockopt(SOCKOPT_SET_IFADDR_SET, &param, sizeof(param));
     case DPIP_CMD_FLUSH:
+        param.ifa_ops = INET_ADDR_FLUSH;
         return dpvs_setsockopt(SOCKOPT_SET_IFADDR_FLUSH, &param, sizeof(param));
     case DPIP_CMD_SHOW:
+        param.ifa_ops = INET_ADDR_GET;
         err = dpvs_getsockopt(SOCKOPT_GET_IFADDR_SHOW, &param, sizeof(param),
                               (void **)&array, &size);
         if (err != 0)
             return err;
 
         if (size < sizeof(*array)
-                || size != sizeof(*array) + \
-                           array->naddr * sizeof(struct inet_addr_param)) {
+                || size < sizeof(*array) + \
+                           array->naddr * sizeof(struct inet_addr_data)) {
             fprintf(stderr, "corrupted response.\n");
```

Comment:
(size < sizeof(*array) || size < sizeof(*array) + \XXXX) means size < sizeof(*array) + \XXXX

Reply:
It make use of  the short-cut evaluation of  operator "||" here.  It's not safe to evaluate the second condition when the first one is not TRUE.